### PR TITLE
add check for PodAntiAffinity not nil

### PR DESCRIPTION
**Pull request by subhamkrai**
add check for PodAntiAffinity not nil before checking
underlying resources otherwise it will cause nil pointer
err. Also updating unit test to test this nil pointer
condition.

Signed-off-by: subhamkrai <srai@redhat.com>

### CODE REVIEW

**Comment by subhamkrai** [comment]
making it draft pr as I'm not able to test in the cluster(struggling with cluster step up)

File: controllers/storagecluster/placement.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookCephv1.Placeme
 
 	topologyKey := getFailureDomain(sc)
 	topologyKey, _ = topologyMap.GetKeyValues(topologyKey)
-	if component == "mon" || component == "mds" || (component == "rgw" && getCephObjectStoreGatewayInstances(sc) > 1) {
+	if (component == "mon" || component == "mds" || (component == "rgw" && getCephObjectStoreGatewayInstances(sc) > 1)) && placement.PodAntiAffinity != nil {
 		if placement.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution != nil {
```
Comment (sp98) [inline controllers/storagecluster/placement.go:66]:
```suggestion
		if placement.PodAntiAffinity != nil && placement.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution != nil {
```

might be easier to read

**Review by sp98 — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/placement.go (hunk 1 above)
Comment (subhamkrai) [reply controllers/storagecluster/placement.go]: changed

**Comment by subhamkrai** [comment]
After taking @agarwal-mudit making it ready for review as this is a small change but I'll also try to test in the local cluster.

**Review by jarrpa — APPROVED** [review]
/lgtm

**Comment by agarwal-mudit** [comment]
@subhamkrai Please create a manual backport PR for release-4.8 as the automated cherry-pick failed because of commits.

**Comment by subhamkrai** [comment]
> @subhamkrai Please create a manual backport PR for release-4.8 as the automated cherry-pick failed because of commits.

#1354
